src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp

```diff
@@ -77,11 +77,25 @@ void FlybyPoint::UpdateState(uint64_t CurrentSimNanos)
         /*! read and allocate the attitude navigation message */
         NavTransMsgPayload relativeState = this->filterInMsg();
 
-        /*! compute radial (ur_N), along-track (ut_N) and out-of-plane (uh_N) unit direction vectors at time of read */
+        /*! compute radial (ur_N), velocity (uv_N), along-track (ut_N), and out-of-plane (uh_N) unit direction vectors */
         double ur_N[3];
         v3Normalize(relativeState.r_BN_N, ur_N);
+        double uv_N[3];
+        v3Normalize(relativeState.v_BN_N, uv_N);
+        double dot = v3Dot(ur_N, uv_N);
         double uh_N[3];
-        v3Cross(relativeState.r_BN_N, relativeState.v_BN_N, uh_N);
+        if (1-dot < this->epsilon) {
+            v3Perpendicular(ur_N, uh_N);
+            this->singularityFlag = 1;
+        }
+        else if (dot + 1 < this->epsilon) {
+            v3Perpendicular(ur_N, uh_N);
+            this->singularityFlag = -1;
+        }
+        else {
+            v3Cross(ur_N, uv_N, uh_N);
+            this->singularityFlag = 0;
+        }
```

Comment:
I would not create the temporary dot variable and just check for 1-dot being less/greater than epsilon inside the ifs in line 114 to 122. That would remove the singularityFlag and would be a bit more legible

Reply:
I've removed the dot variable and combined this if statement into the previous one. I think I still need the singularityFlag, see the reply to your other comment.